src/main/kotlin/com/terraformation/backend/documentproducer/db/VariableStore.kt

```diff
@@ -128,6 +128,23 @@ class VariableStore(
             ?.let { fetchVariableOrNull(it) }
       }
 
+  fun fetchListByStableIds(stableIds: List<StableId>): List<Variable> =
+      with(VARIABLES) {
+        dslContext
+            .select(ID)
+            .from(VARIABLES)
+            .where(STABLE_ID.`in`(stableIds))
+            .and(
+                ID.notIn(
+                    dslContext
+                        .select(REPLACES_VARIABLE_ID)
+                        .from(VARIABLES)
+                        .where(REPLACES_VARIABLE_ID.isNotNull())))
```

Comment:
This subquery is going to fetch every variable that has ever been replaced, whether or not it has one of the requested stable IDs.

Two other ways you could go about this would be to select the maximum variable ID for each stable ID (`DISTINCT ON` would help with that) or to use a correlated `NOT EXISTS` subquery.

Reply:
Ah shoot ya I meant to include that in the subquery.
Updated to the `distinct on` approach (seems that `distinct` is actually not necessary within jooq).